SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -287,6 +289,18 @@ void USpatialReceiver::OnRemoveEntity(const Worker_RemoveEntityOp& Op)
 
 void USpatialReceiver::OnRemoveComponent(const Worker_RemoveComponentOp& Op)
 {
+	// We should exit early if we're receiving a duplicate RemoveComponent op. This can happen with dynamic
+	// components enabled. We detect if the op is a duplicate via the queue of ops to be processed (duplicate
+	// op receive in the same op list) OR whether the data removed from the StaticComponentView (received and
+	// processed in a previous op list).
+	if (QueuedRemoveComponentOps.ContainsByPredicate([&Op](const Worker_RemoveComponentOp& QueuedOp)
+	{
+		return QueuedOp.entity_id == Op.entity_id && QueuedOp.component_id == Op.component_id;
+	}) || !StaticComponentView->HasComponent(Op.entity_id, Op.component_id))
```

Comment:
Worth logging when the second condition fires? Seems rarer, although perhaps that's naive, and might indicate something more sinister going on.